// ngx-iban/esm2022/lib/ngx-iban/iban.directive.mjs
import { NG_VALIDATORS, } from '@angular/forms';
import { Directive, Input } from '@angular/core';
import { electronicFormatIBAN, isValidIBAN } from 'ibantools';
import * as i0 from "@angular/core";
export const ibanValidator = (countryCode) => {
    return (control) => {
        if (countryCode && control.value) {
            return new RegExp(`^${countryCode}.*$`, 'i').test(control.value) &&
                isValidIBAN(electronicFormatIBAN(control.value))
                ? null
                : { iban: { value: control.value } };
        }
        else if (control.value) {
            return isValidIBAN(electronicFormatIBAN(control.value))
                ? null
                : { iban: { value: control.value } };
        }
        return null;
    };
};
export class IbanDirective {
    validate(control) {
        return ibanValidator(this.countryCode)(control);
    }
    /* istanbul ignore next */
    registerOnValidatorChange(fn) {
        this._onChange = fn;
    }
    /* istanbul ignore next */
    ngOnChanges(changes) {
        if ('countryCode' in changes && this._onChange) {
            this._onChange();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.0", ngImport: i0, type: IbanDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "18.0.0", type: IbanDirective, isStandalone: true, selector: "[ngxIban]", inputs: { countryCode: ["ngxIban", "countryCode"] }, providers: [
            { provide: NG_VALIDATORS, useExisting: IbanDirective, multi: true },
        ], usesOnChanges: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.0", ngImport: i0, type: IbanDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[ngxIban]',
                    standalone: true,
                    providers: [
                        { provide: NG_VALIDATORS, useExisting: IbanDirective, multi: true },
                    ],
                }]
        }], propDecorators: { countryCode: [{
                type: Input,
                args: ['ngxIban']
            }] } });
